lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -96,93 +81,83 @@ describe('NxModal', function() {
       }
     });
 
-    const createEvent = (key = 'Escape') => ({
-      key,
-      stopPropagation: jest.fn(),
-      nativeEvent: {
-        stopImmediatePropagation: jest.fn()
-      }
-    });
-
-    it('executes onClose method with a cancel event when pressing ESC key', function () {
-      const mockCallBack = jest.fn();
-      const component = getModal({ onClose: mockCallBack });
+    it('executes onClose method with a cancel event when pressing ESC key', async function () {
+      const mockCallBack = jest.fn(),
+          user = userEvent.setup();
 
+      quickRender({ onClose: mockCallBack });
       expect(mockCallBack).not.toHaveBeenCalled();
-      component.simulate('keyDown', createEvent());
+
+      await user.keyboard('[Escape]');
+
       expect(mockCallBack).toHaveBeenCalledTimes(1);
       expect(mockCallBack.mock.calls[0][0].type).toBe('cancel');
     });
 
-    it('executes onCancel method with a cancel event when pressing ESC key', function () {
-      const mockCallBack = jest.fn();
-      const component = getModal({ onCancel: mockCallBack });
+    it('executes onCancel method with a cancel event when pressing ESC key', async function () {
+      const mockCallBack = jest.fn(),
+          user = userEvent.setup();
 
+      quickRender({ onCancel: mockCallBack });
       expect(mockCallBack).not.toHaveBeenCalled();
-      component.simulate('keyDown', createEvent());
+
+      await user.keyboard('[Escape]');
+
       expect(mockCallBack).toHaveBeenCalledTimes(1);
       expect(mockCallBack.mock.calls[0][0].type).toBe('cancel');
```

Comment:
I saw this in Jest's docs at https://jestjs.io/docs/mock-functions, but curious if you know how this is constructed? I think the first [0] refers to the first call made and the second [0] refers to the argument passed, but I'm just a bit unsure about how it comes up first with `mock.calls`- is there a way in JS/React to parse down into events like that to get the type?

Reply:
> I saw this in Jest's docs at https://jestjs.io/docs/mock-functions, but curious if you know how this is constructed? I think the first [0] refers to the first call made and the second [0] refers to the argument passed, but I'm just a bit unsure about how it comes up first with `mock.calls`- is there a way in JS/React to parse down into events like that to get the type?

I think you're right. `mock.calls` looks to be an array of the mock calls made. The call itself is another array that stores the arguments passed into it. I think this is just the structure that jest follows. I think we can just use the Event interface to get the type.